packages/gatsby-plugin-sitemap/src/internals.js

```diff
@@ -1,123 +1,119 @@
-import fs from "fs"
-import pify from "pify"
-import minimatch from "minimatch"
+import reporter from "gatsby-cli/lib/reporter"
 
+/**
+ *
+ * @param {string} path
+ * @returns {string}
+ */
 export const withoutTrailingSlash = path =>
   path === `/` ? path : path.replace(/\/$/, ``)
 
-export const writeFile = pify(fs.writeFile)
-export const renameFile = pify(fs.rename)
-
-export function filterQuery(
-  results,
-  excludes,
-  pathPrefix,
-  resolveSiteUrl = defaultOptions.resolveSiteUrl
-) {
-  const { errors, data } = results
-
-  if (errors) {
-    throw new Error(errors.join(`, `))
-  }
-
-  const { allSitePage, ...otherData } = data
-
-  let { allPages, originalType } = getNodes(allSitePage)
-
-  // Removing excluded paths
-  allPages = allPages.filter(
-    page =>
-      !excludes.some(excludedRoute =>
-        minimatch(
-          withoutTrailingSlash(page.path),
-          withoutTrailingSlash(excludedRoute)
-        )
-      )
-  )
-
-  // Add path prefix
-  allPages = allPages.map(page => {
-    page.path = (pathPrefix + page.path).replace(/^\/\//g, `/`)
-    return page
-  })
-
-  // siteUrl Validation
+/**
+ * @name prefixPath
+ *
+ * Properly handles prefixing relative path with site domain, Gatsby pathPrefix and AssetPrefix
+ *
+ * @param {string} url - string containing relative path
+ * @param {string} siteUrl - results of the resolveSiteUrl function
+ * @returns {string}
+ */
+// TODO: Update for v3
```

Comment:
What does this todo mean?

Reply:
This has been put in all over Gatsby where we do this janky asset and path prefix handling. I didn't add it but I'm happy to make it more descriptive.

Reply:
please do cause I forgot about it :joy: